src/libsyntax/parse/parser.rs

```diff
@@ -6755,6 +6807,28 @@ impl<'a> Parser<'a> {
                                     maybe_append(attrs, extra_attrs));
             return Ok(Some(item));
         }
+        if self.check_keyword(keywords::Async) &&
+            (self.look_ahead(1, |t| t.is_keyword(keywords::Fn)) ||
+             self.look_ahead(1, |t| t.is_keyword(keywords::Unsafe)))
```

Comment:
The order a lot of people seemed to agree on in the RFC discussion was different. https://github.com/rust-lang/rfcs/pull/2394#discussion_r183191359

"`const unsafe async move`": `unsafe` was in front of `async` back then (of course `move` doesn't apply here) To me `unsafe` in front of `async` seems more "natural" because unsafety should be featured more prominently.